filebeat/input/v2/plugin.go

```diff
@@ -0,0 +1,112 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package v2
+
+import (
+	"github.com/elastic/beats/heartbeat/scheduler"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/go-concert/chorus"
+)
+
+type Plugin struct {
+	Name        string
+	Constraints Constraint
+	Doc         string
+
+	// TODO: config schema info for config validation
+
+	// Configure configures an Input if possible. Returns an error if the
+	// configuration is false.
+	// The input return must not attempt to run or create any kind of connection yet.
+	// The input is only supposed to be a types place holder for the untyped
+	// configuration.
+	Configure func(log *logp.Logger, config *common.Config) (Input, error)
+}
```

Comment:
Can we also add a maturity in the Plugin definition? Stable, Beta, Experimental, we could also automatically add theses context or data to the logger send to the plugin instance.